### PR TITLE
Sub Spaces

### PR DESCRIPTION
**Pull request by mktcode**
closes https://github.com/snapshot-labs/snapshot/issues/1797

Updated the [docs page draft ](https://app.gitbook.com/o/-LFgTZvhAg63US8GVxGf/s/-MG4Ulnnabb2Xz3Lei9_/~/changes/9ZBLxzo5M6OATiLxcAMl/spaces/sub-spaces) but on the other hand it actually feels quite self-explanatory now. Do we really want to maintain an extra docs page for this?

### CODE REVIEW

**Review by alfiomy — APPROVED** [review]

**Review by alfiomy — APPROVED** [review]

File: src/locales/default.json [hunk 1]
```diff
@@ -279,7 +281,20 @@
...
     "hideSpace": "Hide space from homepage",
     "links": "Social accounts",
-    "subspaces": "Sub-Spaces",
+    "subspaces": {
+      "label": "Sub-spaces",
+      "information": "Sub-spaces will only be shown once configured on both the main space and the sub-space(s). {docs}",
+      "parent": {
+        "label": "Main space",
+        "placeholder": "e.g. fabien.eth",
+        "information": "The main space of your DAO, which this space is a sub-space of. It will be displayed in the space's sidebar menu."
```
Comment (samuveth) [inline src/locales/default.json:290]:
```suggestion
        "information": "The space that this space is a sub-space of will be displayed in the sidebar menu"
```

File: src/locales/default.json [hunk 2]
```diff
@@ -279,7 +281,20 @@
...
+      },
+      "children": {
+        "label": "Sub-spaces",
+        "placeholder": "e.g. fabien.eth, balancer.eth, ...",
+        "information": "If this is the main space of your DAO, enter related sub-spaces here. They will be listed in the space's sidebar menu."
```
Comment (samuveth) [inline src/locales/default.json:295]:
```suggestion
        "information": "Enter related sub-spaces here, they will be listed in the space's sidebar menu"
```

**Comment by samuveth** [comment]
Shouldn't they both be above the navigation?

image
image

File: src/components/SpaceSidebarNavigation.vue [hunk 3]
```diff
@@ -22,10 +22,41 @@ const isAdmin = computed(() => {
...
+const subSpaces = computed(() => {
+  return props.space?.children?.filter(
+    space => space.parent?.id === props.space?.id
+  );
+});
 </script>
 
 <template>
   <div>
+    <div v-if="mainSpace" class="mb-3">
```
Comment (samuveth) [inline src/components/SpaceSidebarNavigation.vue:43]: I'd refactor this sub-space logic into a separate component `SpaceSidebarSubspaces.vue`

File: src/locales/default.json (hunk 1 above)
Comment (mktcode) [reply src/locales/default.json:290]: I like. I just don't get why we're not using full-stops to end a sentence anymore. I've seen this in other places to.

**Comment by mktcode** [comment]
> Shouldn't they both be above the navigation?

For the main space it made more sense to me. Sub-spaces can be up to 8, pushing the menu down quite a bit. I'd rather put both below the menu.

File: src/components/SpaceSidebarNavigation.vue (hunk 3 above)
Comment (mktcode) [reply src/components/SpaceSidebarNavigation.vue:43]: Yea I wanted to do that... but... but... well... :D

**Comment by hjjlxm** [comment]
Speaking of self-explanatory, I propose another solution, which I believe makes the UI more simple. @mktcode @samuveth 
Space profile

**Comment by samuveth** [comment]
@hjjlxm thanks, I will forward this to our design team 👍

**Pull request merged**